src/services/depositService.ts

```diff
@@ -0,0 +1,245 @@
+ *
+ * @param userAddress - The user address from the wallet
+ * @param chain - The blockchain for which to generate the address
+ * @returns A Promise that resolves to the generated deposit address
+ */
+export async function generateDepositAddress(
+  userAddress: DefuseUserId,
+  chain: BlockchainEnum
+): Promise<string> {
+  try {
+    const supportedTokens = await getSupportedTokens({
+      chains: [chain],
+    })
+
+    if (supportedTokens.tokens.length === 0) {
+      throw new Error("No supported tokens found")
+    }
+
+    const generatedDepositAddress = await getDepositAddress({
+      account_id: userAddress,
+      chain,
+    })
+
+    return generatedDepositAddress.address
+  } catch (error) {
+    console.error("Error generating deposit address:", error)
+    throw error
+  }
+}
+
+export async function checkNearTransactionValidity(
+  txHash: string,
+  accountId: string,
+  amount: string
+): Promise<boolean> {
+  if (!txHash) {
+    throw new Error("Transaction hash is required")
+  }
+  const successValue = await getNearTxSuccessValue({
+    txHash,
+    senderAccountId: accountId,
+  })
+  // Check if input amount is equal to the success value
+  return successValue === BigInt(amount)
+}
+
+export async function isStorageDepositRequired(
+  contractId: string,
+  accountId: string
+): Promise<boolean> {
+  // Aurora is a special case and does not require storage deposit
+  if (contractId === "aurora") {
+    return false
+  }
+  const storageBalance = await getNearNep141StorageBalance({
+    contractId,
+    accountId,
+  })
+  return storageBalance < (await getMinimumStorageBalance(contractId))
+}
+
+export async function getMinimumStorageBalance(
+  contractId: string
+): Promise<bigint> {
+  return getNearNep141MinStorageBalance({
+    contractId,
+  })
+}
+
+/**
+ * Get the available deposit routes for a given wallet connection and selected network.
+ *
+ * @param chainTypeFromWallet - The type of chain from wallet connection [near, evm].
+ * @param network - The network to check.
+ * @returns An object containing the available deposit routes.
+ *
+ * @remarks
+ * - `activeDeposit` is deposit via wallet extension.
+ * - `passiveDeposit` is deposit via generated address at QR code provided by POA bridge.
+ */
+export function getAvailableDepositRoutes(
+  chainTypeFromWallet: ChainType,
+  network: BlockchainEnum
+): { activeDeposit: boolean; passiveDeposit: boolean } | null {
+  switch (chainTypeFromWallet) {
+    case ChainType.Near:
+      switch (network) {
+        case BlockchainEnum.NEAR:
+          return {
+            activeDeposit: true,
+            passiveDeposit: false,
+          }
+        case BlockchainEnum.ETHEREUM:
+        case BlockchainEnum.BASE:
+        case BlockchainEnum.ARBITRUM:
+        case BlockchainEnum.BITCOIN:
+          return {
+            activeDeposit: false,
+            passiveDeposit: true,
+          }
+        default:
+          return null
```

Comment:
Suggest adding exhaustive check, so if you add new blockchain then it would yell here.